### PR TITLE
Workflow and playwright test updates

### PR DESCRIPTION
**Pull request by ytimocin**
1. Added Playwright failure videos.
2. Removing `Wait for all pods to be ready` step because the previous step, `rad deploy`, should do the waiting already.
3. Making sure we skip certain steps if `Cancel Workflow` is called.
4. Adding the name of the EKS Cluster to the `Delete EKS Cluster` step. (#1003)
5. Removing `rad uninstall` from the `Delete EKS Cluster` step. (#868)
6. Updating the dependencies in the playwright **package.json** file.
7. Changing to liveness check instead of readiness check. Please see this [comment](https://github.com/radius-project/samples/pull/1360/files#r1580358246).
8. Added a logic to reload page if the catalog items are not found. Not sure if this is an eShop problem but sometimes the service that needs to return the products throws an error. Please see this [video](https://teams.microsoft.com/l/message/19:ve3Qj3hke37Z4b1EdjJy-K4nzh3nDgRxZJJARwusRgc1@thread.tacv2/1714115452205?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=b33db407-e89b-4fb6-87cc-9b2f9f8ec265&parentMessageId=1711137700692&teamName=Radius%20Core%20Team&channelName=General&createdTime=1714115452205).

Fixes:
#1003 
#868 
#1017

### CODE REVIEW

File: samples/eshop/services/webstatus.bicep [hunk 1]
```diff
@@ -25,25 +24,25 @@ resource webstatus 'Applications.Core/containers@2023-10-01-preview' = {
...
+        HealthChecksUI__HealthChecks__7__Uri: 'http://identity-api:5105/liveness'
         HealthChecksUI__HealthChecks__8__Name: 'Payments HTTP Check'
-        HealthChecksUI__HealthChecks__8__Uri: 'http://payment-api:5108/hc'
+        HealthChecksUI__HealthChecks__8__Uri: 'http://payment-api:5108/liveness'
         HealthChecksUI__HealthChecks__9__Name: 'Ordering SignalRHub HTTP Check'
-        HealthChecksUI__HealthChecks__9__Uri: 'http://ordering-signalrhub:5112/hc'
+        HealthChecksUI__HealthChecks__9__Uri: 'http://ordering-signalrhub:5112/liveness'
         HealthChecksUI__HealthChecks__10__Name: 'Ordering HTTP Background Check'
-        HealthChecksUI__HealthChecks__10__Uri: 'http://ordering-backgroundtasks:5111/hc'
+        HealthChecksUI__HealthChecks__10__Uri: 'http://ordering-backgroundtasks:5111/liveness'
```
Comment (willdavsmith) [inline samples/eshop/services/webstatus.bicep:45]: were we using the health check incorrectly before?

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -256,39 +256,33 @@ jobs:
...
-            npx playwright install --with-deps
-            npx playwright test ${{ matrix.uiTestFile }} --retries 3
+          command: rad deploy ${{ matrix.path }} ${{ matrix.deployArgs }}
+      - name: Run Playwright Test
+        if: steps.gen-id.outputs.RUN_TEST == 'true' && matrix.uiTestFile != ''
+        id: run-playwright-test
+        run: |
+          if [[ "${{ matrix.container }}" != "" ]]; then
+            rad resource expose containers ${{ matrix.container }} ${{ matrix.exposeArgs }} --port ${{ matrix.port }} &
+            export ENDPOINT="http://localhost:3000/"
```
Comment (willdavsmith) [inline .github/workflows/test.yaml:275]:
```suggestion
            export ENDPOINT="http://localhost:${{ matrix.port }}"
```

File: samples/eshop/services/webstatus.bicep (hunk 1 above)
Comment (rynowak) [reply samples/eshop/services/webstatus.bicep:45]: The eshop code has a readiness check at `/hc` and a liveness check at `/liveness`. We had the readiness check hooked up as a liveness check

File: .github/workflows/test-aks.yaml [hunk 3]
```diff
@@ -22,6 +22,9 @@ on:
       - edge
   schedule: # Run every 2 hours
     - cron: "0 */2 * * *"
+concurrency:
```
Comment (rynowak) [inline .github/workflows/test-aks.yaml]: Can you add a comment explaining why this is needed? It's easy to forget the rationale for adding something like this later.

File: .github/workflows/test-aks.yaml [hunk 4]
```diff
@@ -283,14 +286,12 @@ jobs:
       - name: Deploy app
         if: steps.gen-id.outputs.RUN_TEST == 'true'
         id: deploy-app
-        run: rad deploy ${{ matrix.path }} ${{ matrix.deployArgs }} -e ${{ matrix.env }}
-      - name: Wait for all pods to be ready
```
Comment (rynowak) [inline .github/workflows/test-aks.yaml:287]: Do we no longer care about waiting for the pods to be ready?

**Comment by rynowak** [comment]
Is this working better now that you swapped the health checks?

File: .github/workflows/test-aks.yaml (hunk 4 above)
Comment (ytimocin) [reply .github/workflows/test-aks.yaml:287]: I think we discussed this in one of our 1:1 meeting. `rad deploy` should wait for all pods to be ready before returning successfully. Is that wrong?

**Comment by ytimocin** [comment]
> Is this working better now that you swapped the health checks?

Yes, it definitely is working better. Not passing the tests all the time but still a good improvement so far with the changes here.

File: .github/workflows/test-aks.yaml (hunk 3 above)
Comment (ytimocin) [reply .github/workflows/test-aks.yaml]: This is to group workflow runs and cancel if a new workflow starts running. What I mean is that, if we create a PR, a workflow is triggered. Then if we push another commit to that PR, another workflow will be triggered and the in-progress one should be cancelled.


Now that I think about this, because we might have already created resources, I think it would be good to not cancel this workflow and make sure that the resources are deleted if created.

Comment (ytimocin) [reply .github/workflows/test-aks.yaml]: What do you think @rynowak?

File: .github/scripts/delete-aws-resources.sh [hunk 5]
```diff
@@ -34,21 +33,16 @@ function delete_aws_resources() {
   # Empty the file
   truncate -s 0 $DELETED_RESOURCES_FILE
 
-  for resource_type in ${RESOURCE_TYPES//,/ }
```
Comment (willdavsmith) [inline .github/scripts/delete-aws-resources.sh:37]: why the changes to this file?

**Review by willdavsmith — APPROVED** [review]

File: .github/scripts/delete-aws-resources.sh (hunk 5 above)
Comment (ytimocin) [reply .github/scripts/delete-aws-resources.sh:37]: just a reformat
